ui/tx/assetFlows/components/NovesActionSnippet.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Hide, Popover, PopoverArrow, PopoverContent, PopoverTrigger, Show, Skeleton, Text, useColorModeValue } from '@chakra-ui/react';
+import type { FC } from 'react';
+import React from 'react';
+
+import TokenEntity from 'ui/shared/entities/token/TokenEntity';
+import IconSvg from 'ui/shared/IconSvg';
+
+import type { NovesFlowViewItem } from '../utils/generateFlowViewData';
+import NovesTokenTooltipContent from './NovesTokenTooltipContent';
+
+interface Props {
+  item: NovesFlowViewItem;
+  isLoaded: boolean;
+}
+
+const NovesActionSnippet: FC<Props> = ({ item, isLoaded }) => {
+  const popoverBg = useColorModeValue('gray.700', 'gray.300');
+
+  const token = React.useMemo(() => {
+    const action = item.action;
+
+    const name = action.nft?.name || action.token?.name;
+    const symbol = action.nft?.symbol || action.token?.symbol;
+
+    const token = {
+      name: name,
+      symbol: symbol?.toLowerCase() === name?.toLowerCase() ? undefined : symbol,
+      address: action.nft?.address || action.token?.address,
+    };
+
+    return token;
+  }, [ item.action ]);
+
+  const validTokenAddress = token.address ? Boolean(token.address.match(/^0x[a-fA-F\d]{40}$/)) : false;
```

Comment:
We already have the regular expression for address hash. Please refer to `lib/regexp.ts`.